src/note.rs

```diff
@@ -108,7 +110,8 @@ impl PartialEq for Note {
 impl Eq for Note {}
 
 impl Note {
-    pub(crate) fn from_parts(
+    /// Construct note from its component parts
+    pub fn from_parts(
```

Comment:
This doesn't check the invariant `note.commitment_inner().is_some()`. It can be made public provided it checks that invariant (i.e. it would return `CtOption<Self>`).

There are two existing uses of `Note::from_parts` that need to be changed in that case:

https://github.com/zcash/orchard/blob/3faab98e9e82618a0f2d887054e9e28b0f7947dd/src/note_encryption.rs#L78
I think the correct change here is for the whole `orchard_parse_note_plaintext_without_memo` function to return `None` if `Note::from_parts` does. Any change to this code is consensus-critical and requires three reviews (at least two from ECC engineers).

----
https://github.com/zcash/orchard/blob/3faab98e9e82618a0f2d887054e9e28b0f7947dd/src/keys.rs#L1047-L1052
This is in a test so can just unwrap.